### PR TITLE
Update libtiledb version in readthedocs config

### PR DESCRIPTION
**Pull request by kounelisagis**
Also fixes: `WARNING: Invalid configuration value found: 'language = None'. Update your configuration to a valid language code. Falling back to 'en' (English).`

### CODE REVIEW

**Review by ihnorton — CHANGES_REQUESTED** [review]

**Review by ihnorton — APPROVED** [review]
LGTM but please try setuptools_scm before merging so we don't have to update manually.

**Comment by ihnorton** [comment]
> (There are many more things they suggest for modernizing the setup.py)

Big +1 to cleaning this up, great observation. It's been on my list to discuss w/ you soon 😉. We can probably cut the size of the file by 95% by using [scikit-build-core](https://scikit-build-core.readthedocs.io/en/latest/) -- see how it's used in [vector search](https://github.com/TileDB-Inc/TileDB-Vector-Search/blob/main/apis/python/CMakeLists.txt) to handle the extension build, and then the modernization should be really straightforward.

**Review by ihnorton — COMMENTED** [review]
Overall LGTM, but could you please do a test build by pushing this branch as `release-test-sc35414`? Then after ~1hr you should have wheels available [here](https://dev.azure.com/TileDB-Inc/CI/_build?definitionId=1) (just invited you to project). Make sure one of the wheels installs and `import`s correctly locally, and that the metadata looks the same as the last release on PyPI.

File: pyproject.toml [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
+maintainers = [
+  {name = "TileDB, Inc.", email = "help@tiledb.io"}
+]
+license = {text = "MIT"}
+dynamic = ["version"]
+
+[project.urls]
+homepage = "https://github.com/TileDB-Inc/TileDB-Py"
+
+[tool.setuptools_scm]
```
[the remark this answers is not in the document]
Comment (kounelisagis) [reply pyproject.toml:21]: This is an additional step: https://pypi.org/project/setuptools-scm/

[the remark this answers is not in the document]
Comment (kounelisagis) [reply pyproject.toml:41]: We use them in:

```
setup(
    name="tiledb",
    platforms=["any"],
    use_scm_version={
        "version_scheme": "guess-next-dev",
        "local_scheme": "dirty-tag",
        "write_to": "tiledb/_generated_version.py",
    },
...
```

**Comment by kounelisagis** [comment]
> Overall LGTM, but could you please do a test build by pushing this branch as `release-test-sc35414`? Then after ~1hr you should have wheels available [here](https://dev.azure.com/TileDB-Inc/CI/_build?definitionId=1) (just invited you to project). Make sure one of the wheels installs and `import`s correctly locally, and that the metadata looks the same as the last release on PyPI.

I used tiledb-0.26.3-cp311-cp311-macosx_11_0_arm64.whl file found in **drop** folder.

```
conda create -n test_env python=3.11
conda activate test_env
pip install tiledb-0.26.3-cp311-cp311-macosx_11_0_arm64.whl
pip install numpy
python
>>> import tiledb
>>> from importlib import metadata
>>> dict(metadata.metadata('tiledb'))
{'Metadata-Version': '2.1', 'Name': 'tiledb', 'Version': '0.26.3', 'Summary': 'Pythonic interface to the TileDB array storage manager', 'Author-email': '"TileDB, Inc." <help@tiledb.io>', 'Maintainer-email': '"TileDB, Inc." <help@tiledb.io>', 'License': 'MIT', 'Project-URL': 'homepage, https://github.com/TileDB-Inc/TileDB-Py', 'Platform': 'any', 'Description-Content-Type': 'text/markdown', 'License-File': 'LICENSE', 'Description': '<a href="https://tiledb.com/"><img src="https://github.com/TileDB-Inc/TileDB/raw/dev/doc/source/_static/tiledb-logo_color_no_margin_@4x.png" alt="TileDB logo" width="400"></a>\n\n\n[![Build Status](https://dev.azure.com/TileDB-Inc/CI/_apis/build/status/TileDB-Inc.TileDB-Py?branchName=dev)](https://dev.azure.com/TileDB-Inc/CI/_build/latest?definitionId=1&branchName=dev)\n![](https://raw.githubusercontent.com/TileDB-Inc/TileDB/dev/doc/anaconda.svg?sanitize=true)[![Anaconda download count badge](https://anaconda.org/conda-forge/TileDB-Py/badges/downloads.svg)](https://anaconda.org/conda-forge/TileDB-Py)\n\n\n# TileDB-Py\n\n*TileDB-Py* is a [Python](https://python.org/) interface to the [TileDB Storage Engine](https://github.com/TileDB-Inc/TileDB).\n\n# Quick Links\n\n* [Installation](https://docs.tiledb.com/developer/installation/quick-install)\n* [Build Instructions](https://docs.tiledb.com/main/how-to/installation/building-from-source/python)\n* [TileDB Documentation](https://docs.tiledb.com/main/)\n* [Python API reference](https://tiledb-inc-tiledb-py.readthedocs-hosted.com/en/stable)\n\n# Quick Installation\n\nTileDB-Py is available from either [PyPI](https://pypi.org/project/tiledb/) with ``pip``:\n\n
\npip install tiledb\n```\n\nor from conda-forge with\nconda or mamba:\n\n```\nconda install -c conda-forge tiledb-py\n```\n\nDataframes functionality (tiledb.from_pandas, Array.df[]) requires Pandas 1.0 or higher, and PyArrow 1.0 or higher.\n\n# Contributing\n\nWe welcome contributions, please see [CONTRIBUTING.md](CONTRIBUTING.md) for suggestions and\ndevelopment-build instructions. For larger features, please open an issue to discuss goals and\napproach in order to ensure a smooth PR integration and review process.\n'}
```

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dudoslav) [reply pyproject.toml:41]: Ah, so this kind of concerns my last comment about having two separate files and variables across them. Never mind, fine by me for this PR.

**Review by dudoslav — APPROVED** [review]
LGTM, just make sure that you can still produce wheel, so please:

1. Create and install wheel
2. Create and install sdist
3. In all cases run `import tiledb` in python

**Comment by kounelisagis** [comment]
> LGTM, just make sure that you can still produce wheel, so please:
> 
> 1. Create and install wheel
> 2. Create and install sdist
> 3. In all cases run `import tiledb` in python

1. Successfully installed wheel from azure devops.
2. sdist
```
> python --version
Python 3.8.17
> pipx run build --sdist
> pip install dist/tiledb-0.26.4.dev3+dirty.tar.gz
> cd ..
> python
>>> import tiledb
>>> tiledb.version()
(0, 26, 4, 'dev3', 'dirty')
>>> tiledb.libtiledb.version()
(2, 20, 1)
```

**Pull request merged**